burn-import/src/onnx/from_onnx.rs

```diff
@@ -428,10 +431,87 @@ fn move_inputs_to_state(nodes: &mut Vec<Node>, initializer: &[TensorProto]) {
             .collect();
 
         // Set the node's states vector to the temporary node_states vector
-        node.states = node_states;
+        node.states.append(&mut node_states);
     });
 }
 
+/// Lift constants from the graph into the states vector for known node types.
+///
+/// The main reason to move constants into the states vector is to reduce the number of nodes in the graph,
+/// and consistently use the same interface for all nodes (constant inputs and inputs with initializers are
+/// treated the same way).
+///
+/// # Arguments
+///
+/// * `nodes` - A mutable reference to a vector of nodes
+///
+/// # Panics
+///
+/// Panics if the node's output is not a constant.
```

Comment:
I think an example would help here to understand what the function actually do (how does the graph look before and after).